kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -140,8 +140,9 @@ class SparkProcessBuilder(
       allConf = allConf ++ zkAuthKeytabFileConf(allConf)
     }
     // pass spark engine log path to spark conf
-    (allConf ++ engineLogPathConf ++ appendPodNameConf(allConf)).foreach { case (k, v) =>
-      buffer ++= confKeyValue(convertConfigKey(k), v)
+    (allConf ++ engineLogPathConf ++ extraYarnConf ++ appendPodNameConf(allConf)).foreach {
```

Comment:
It looks like `extraYarnConf` will override the user's configuration.

Reply:
I have moved extraYarnConf to the front

Reply:
Emm... if we really want to respect user's conf here, we need to do things like `appendPodNameConf` does

Reply:
thanks, fixed

Reply:
The construction of allConf is too complex, I prefer to put kvs into a mutable map, let's optimize it later.